src/ert/_c_wrappers/enkf/ert_workflow_list.py

```diff
@@ -1,190 +1,89 @@
-import ctypes
+import logging
 import os
 from typing import Iterator, List
 
-from cwrap import BaseCClass
-
-from ert._c_wrappers import ResPrototype
+from ert._c_wrappers.config.config_parser import ConfigValidationError
 from ert._c_wrappers.enkf.config_keys import ConfigKeys
 from ert._c_wrappers.enkf.enums import HookRuntime
-from ert._c_wrappers.enkf.hook_workflow import HookWorkflow
 from ert._c_wrappers.job_queue import Workflow, WorkflowJob, WorkflowJoblist
-from ert._c_wrappers.util.substitution_list import SubstitutionList
-
-
-def _to_c_string_arr(lst: List[str]):
-    arr = (ctypes.c_char_p * len(lst))()
-    arr[:] = [s.encode("utf-8") for s in lst]
-    return arr
-
-
-class ErtWorkflowList(BaseCClass):
-    TYPE_NAME = "ert_workflow_list"
-    _alloc = ResPrototype(
-        "void* ert_workflow_list_alloc(config_content, config_content)",
-        bind=False,
-    )
-    _alloc_full = ResPrototype(
-        "void* ert_workflow_list_alloc_full(workflow_joblist)", bind=False
-    )
-    _free = ResPrototype("void ert_workflow_list_free(ert_workflow_list)")
-    _alloc_namelist = ResPrototype(
-        "stringlist_obj ert_workflow_list_alloc_namelist(ert_workflow_list)"
-    )
-    _has_workflow = ResPrototype(
-        "bool ert_workflow_list_has_workflow(ert_workflow_list, char*)"
-    )
-    _get_workflow = ResPrototype(
-        "workflow_ref ert_workflow_list_get_workflow(ert_workflow_list, char*)"
-    )
-    _add_workflow = ResPrototype(
-        "workflow_ref ert_workflow_list_add_workflow(ert_workflow_list, char*, char*)"
-    )
-    _add_job = ResPrototype(
-        "void ert_workflow_list_add_job(ert_workflow_list, char*, char*)"
-    )
-    _has_job = ResPrototype("bool ert_workflow_list_has_job(ert_workflow_list, char*)")
-    _get_job = ResPrototype(
-        "workflow_job_ref ert_workflow_list_get_job(ert_workflow_list, char*)"
-    )
-    _get_job_names = ResPrototype(
-        "stringlist_obj ert_workflow_list_get_job_names(ert_workflow_list)"
-    )
-
-    _get_hook_workflow = ResPrototype(
-        "hook_workflow_obj ert_workflow_list_iget_hook_workflow(ert_workflow_list, int)"
-    )
-
-    _num_hook_workflows = ResPrototype(
-        "int ert_workflow_list_num_hook_workflows(ert_workflow_list)"
-    )
 
+logger = logging.getLogger(__name__)
+
+
+class ErtWorkflowList:
     def __init__(
         self,
-        config_content=None,
+        content_dict=None,
```

Comment:
Instead of passing dicts to `__init__`, I would prefer it if we pass `workflow_job_info`, `workflow_job_dir_info`, `hook_workflow_info` and `workflow_info` to `__init__` instead and create a classmethod `from_dict` which gets the dictionary and creates the class via `__init__`. See `analysis_config.py` for an example of what I mean.